tests/test_transport.py

```diff
@@ -164,7 +174,11 @@ def test_transport_works(
     out, err = capsys.readouterr()
     assert not err and not out
     assert capturing_server.captured
-    assert capturing_server.captured[0].compressed == (compression_level > 0)
+    assert (
+        capturing_server.captured[0].compressed == (compression_level is None)
+        or (compression_level > 0)
+        or (compression_algo is None)
```

Comment:
Can we maybe write this in a less condensed way so that it's a bit easier to grok?

Reply:
Okay, you actually caught both a test logic error _and_ the handling logic around invalid compression algo case.